python/tank/util/shotgun.py

```diff
@@ -897,6 +901,7 @@ def _create_published_file(tk, context, path, name, version_number, task, commen
         "task": task,
         "version_number": version_number,
     }
+    data.update(sg_fields)
```

Comment:
It would be good to move this so it happens before the default values get set. This way we won't allow `sg_fields` to overwrite any of the standard parameter values. Something like:

```
data = {}
# we set the optional additional fields first so we don't allow overwriting the standard parameters!
data.update(sg_fields)

# standard parameters
data["code"] = code,
data["description"] = comment,
...
```